homeassistant/helpers/config_validation.py

```diff
@@ -518,18 +541,85 @@ def ensure_list_csv(value: Any) -> Sequence:
     return ensure_list(value)
 
 
-def deprecated(key):
-    """Log key as deprecated."""
+def deprecated(key: str,
+               replacement_key: Optional[str] = None,
+               invalidation_version: Optional[str] = None,
+               default: Optional[Any] = None):
+    """
+    Log key as deprecated and provide a replacement (if exists).
+
+    Expected behavior:
+        - Outputs the appropriate deprecation warning if key is detected
+        - Processes schema moving the value from key to replacement_key
+        - Processes schema changing nothing if only replacement_key provided
+        - No warning if only replacement_key provided
+        - No warning if neither key nor replacement_key are provided
```

Comment:
I was trying to say that they were not provided within the config being validated, key has to be provided to the deprecated validator. This would happen if we had an optional key which we deprecated and therefore we want to shift the default from the deprecated optional key to the replacement_key.